modules/arcgis/src/commons.ts

```diff
@@ -17,7 +16,43 @@ export type RenderResources = {
   fbo: Framebuffer;
 };
 
-export function initializeResources(this: Renderer, device: Device): RenderResources {
+async function createDeckInstance(gl: WebGL2RenderingContext): Promise<{
+  deckInstance: Deck;
+  device: Device;
+}> {
+  return new Promise(resolve => {
+    const deckInstance = new Deck({
+      // The view state will be set dynamically to track the MapView current extent.
+      viewState: {},
+
+      // Input is handled by the ArcGIS API for JavaScript.
+      controller: false,
+
+      // We use the same WebGL context as the ArcGIS API for JavaScript.
+      gl,
+
+      // We need depth testing in general; we don't know what layers might be added to the deck.
+      parameters: {
+        depthCompare: 'less-equal'
+      },
+
+      // To disable canvas resizing, since the FBO is owned by the ArcGIS API for JavaScript.
+      width: null,
+      height: null,
+
+      onDeviceInitialized: (device: Device) => {
+        resolve({deckInstance, device});
+      }
+    });
+  });
+}
+
+export async function initializeResources(
+  this: Renderer,
+  gl: WebGL2RenderingContext
+): Promise<RenderResources> {
+  const {deckInstance, device} = await createDeckInstance(gl);
```

Comment:
@ibgreen Can't use `luma.attachDevice` here because webgl is registered inside Deck's constructor

Reply:
Since the arcgis module only works on WebGL it wouldn't seem unreasonable to import webgl module here and register it. 

Alternatively, I am not sure if `props.devices` on `luma.attachDevice({..., devices: [WebGLDevice])` works well yet, but that would avoid registration/global mechanisms.

Reply:
The bug here is caused by the webgl module depending on some luma.gl/core exports that deck.gl/core is not exposing.

Reply:
OK that should hopefully be solved in 9.1.